capsules/src/usb/ctap.rs

```diff
@@ -0,0 +1,401 @@
+                transfer_type: TransferType::Interrupt,
+                max_packet_size: 64,
+                interval: 5,
+            },
+            EndpointDescriptor {
+                endpoint_address: EndpointAddress::new_const(0x01, TransferDirection::HostToDevice),
+                transfer_type: TransferType::Interrupt,
+                max_packet_size: 64,
+                interval: 5,
+            },
+        ]];
+
+        let (device_descriptor_buffer, other_descriptor_buffer) =
+            descriptors::create_descriptor_buffers(
+                descriptors::DeviceDescriptor {
+                    vendor_id: vendor_id,
+                    product_id: product_id,
+                    manufacturer_string: 1,
+                    product_string: 2,
+                    serial_number_string: 3,
+                    class: 0x03, // Class: HID
+                    max_packet_size_ep0: MAX_CTRL_PACKET_SIZE,
+                    ..descriptors::DeviceDescriptor::default()
+                },
+                descriptors::ConfigurationDescriptor {
+                    ..descriptors::ConfigurationDescriptor::default()
+                },
+                interfaces,
+                endpoints,
+                Some(&HID_DESCRIPTOR),
+                None,
+            );
+
+        CtapHid {
+            client_ctrl: ClientCtrl::new(
+                controller,
+                device_descriptor_buffer,
+                other_descriptor_buffer,
+                Some(&HID_DESCRIPTOR),
+                Some(&REPORT),
+                LANGUAGES,
+                strings,
+            ),
+            buffers: [Buffer64::default(), Buffer64::default()],
+            client: OptionalCell::empty(),
+            send_buffer: TakeCell::empty(),
+            recv_buffer: TakeCell::empty(),
+            recv_len: Cell::new(0),
+            recv_offset: Cell::new(0),
+            saved_endpoint: OptionalCell::empty(),
+        }
+    }
+
+    #[inline]
+    fn controller(&self) -> &'a U {
+        self.client_ctrl.controller()
+    }
+
+    #[inline]
+    fn buffer(&'a self, i: usize) -> &'a [VolatileCell<u8>; 64] {
+        &self.buffers[i - 1].buf
+    }
+
+    pub fn set_client(&'a self, client: &'a dyn hil::usb_hid::Client<'a>) {
+        self.client.set(client);
+    }
+
+    fn can_receive(&'a self) -> bool {
+        self.client
+            .map(move |client| client.can_receive())
+            .unwrap_or(false)
+    }
+}
+
+impl<'a, U: hil::usb::UsbController<'a>> hil::usb_hid::UsbHid<'a> for CtapHid<'a, U> {
+    fn send_buffer(
+        &'a self,
+        send: &'static mut [u8; 64],
+    ) -> Result<usize, (ReturnCode, &'static mut [u8; 64])> {
+        let len = send.len();
+
+        self.send_buffer.replace(send);
+        self.controller().endpoint_resume_in(ENDPOINT_IN_NUM);
+
+        Ok(len)
+    }
+
+    fn receive_buffer(
+        &'a self,
+        recv: &'static mut [u8; 64],
```

Comment:
Why would the buffers need to be with a static lifetime? Also applies to `send_buffer` above.

Reply:
This is what the other Tock HILs do as well.